static/js/hotkey.js

```diff
@@ -662,6 +670,16 @@ export function process_hotkey(e, hotkey) {
         return true;
     }
 
+    if (event_name === "toggle_edit_preview" && compose_state.composing()) {
+        e.preventDefault();
+
+        if ($("#compose .markdown_preview").css("display") != "none") {
+            compose.show_preview_area();
+        } else {
+            compose.clear_preview_area();
+        }
+    }
```

Comment:
I think this logic belongs with the other `compose.js` `handle_keydown` handlers, not here.  Also `compose_state.composing()` doesn't tell us whether focus is in the compose box, which is what we actually want to know.

Reply:
I don't think that this belongs to `handle_keydown`, the reason being when we are in in the preview state, that `handle_keydown` will not work. 
Also regarding the focus, the compose box is open whenever `compose_state.compising()` is true. And as soon as the user shifts the focus to something, else. The compose box is closed. 
Also I feel that instead of checking if the focus is on the composebox or not. The user will be able to get a better idea when `ctrl+shift+P` will work in case of compose box is open.